### PR TITLE
Explains why the config file is useful.

### LINKED ISSUE #315: sw-cli: consider explaining why saving to a config file is useful
**Library Affected**:
sw-cli

Users might find it useful to understand why saving to a config file is a good idea:
image

### PR DESCRIPTION
**Pull request by jpmedley**
R: @jeffposnick @addyosmani @gauntface

Fixes #315

### CODE REVIEW

**Comment by jpmedley** [comment]
This doesn't look like my message. Am I missing something?

```[Failed] broadcast manager test > check broadcast
    Timeout of 100ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
    [Stack Trace]
    testReceiverChannel.onmessage@http://localhost:3004/packages/sw-background-sync-queue/test/browser/broadcast-manager.js:32:4```

**Comment by gauntface** [comment]
you've expressed concern over the config file, how it should work and how it should be altered - should we wait until that or do you want this landed regardless?

**Comment by jpmedley** [comment]
I withdrew my concern. I think it came about because of a work interruption that made me forget what I was doing.

**Comment by jpmedley** [comment]
There's a note to that affect in #315

File: docs/index.md [hunk 1]
```diff
@@ -15,8 +15,9 @@ offline analytics, background syncing, and more.
 
 ## A service worker in two minutes
 
-Use [sw-cli](sw-cli)
-to add a service worker to an existing project in under two minutes.
+Use [sw-cli](sw-cli) to add a service worker to an existing project in under two
+minutes. Save your cli responses to a configuration file so that updates are
```
Comment (gauntface) [inline docs/index.md]: Just my two cents - this config file bit feels a bit too much info for the developer.

File: docs/sw-cli.md [hunk 2]
```diff
@@ -36,6 +36,20 @@ After the command has run you'll have two new files, a
 `sw-lib.vX.X.X.min.js` and `sw.js` file (unless you changed
 the file name).
 
+## Creating a Config File
+
+One of the questions asked by `sw-cli generate:sw` is whether you'd like to save
+your responses to a configuration file. Answering `Y` will save your responses
+to `sw-cli-config.json`. Subsequent runs of the cli will regenerate the service
+worker based on the config file. Not only does this make building faster, it
+opens the possibility of running subsequent calls in a build script.
```
Comment (gauntface) [inline docs/sw-cli.md]: Maybe make this a little more explicit "opens the possibility of running the CLI without needing user input, like running from a build script".

File: docs/sw-cli.md [hunk 3]
```diff
@@ -36,6 +36,20 @@ After the command has run you'll have two new files, a
...
+
+For example, if you're using gulp you can pass `sw-cli generate:sw` to
```
Comment (gauntface) [inline docs/sw-cli.md]: I think developers using gulp would find it easier to use sw-build @jeffposnick @addyosmani thoughts on this?

File: docs/index.md (hunk 1 above)
Comment (jpmedley) [inline docs/index.md]: I felt like it reinforced the idea of a simple workflow. I'd like some other opinions before removing it.

File: docs/sw-cli.md (hunk 3 above)
Comment (jpmedley) [inline docs/sw-cli.md]: Is that a false choice? Seems like both options might have their place.

Comment (addyosmani) [inline docs/sw-cli.md]: I think both options have their place but that using sw-build would probably address the 90% case without them having to think about cli. Perhaps we steer them towards build and mention the cli workflow here as an alternative?

Comment (jpmedley) [inline docs/sw-cli.md]: Removing text from docs/index.md.

**Comment by jpmedley** [comment]
@gauntface @addyosmani Take another look, guys.

**Comment by jpmedley** [comment]
Ping!

File: docs/sw-cli.md [hunk 4]
```diff
@@ -36,6 +36,21 @@ After the command has run you'll have two new files, a
...
 the file name).
 
+## Creating a Config File
+
+One of the questions asked by `sw-cli generate:sw` is whether you'd like to save
+your responses to a configuration file. Answering `Y` will save your responses
+to `sw-cli-config.json`. Subsequent runs of the cli will regenerate the service
+worker based on the config file, which makes building faster. 
+
+If you don't need anything more complicated (in other words if you don't need
```
Comment (jeffposnick) [inline docs/sw-cli.md]: I don't see using a `gulp` plugin that shells out to `sw-cli` as being a build process that we'd recommend. This paragraph could just be cut.

There's also no reason to assume that someone reading the docs for `sw-cli` is using `gulp`, so if you did feel like this paragraph needs to stay, I wouldn't make it `gulp`-specific.

Comment (jpmedley) [inline docs/sw-cli.md]: @jeffposnick Removed. Please review.

**Pull request merged**